classes/PluginModelItemObject.php

```diff
@@ -718,19 +911,70 @@ protected function getImageInfo($imageId)
 	 */
 	protected function getTierPricesFromDb()
 	{
-		$select = sprintf(
-			'SELECT * from %sspecific_price WHERE id_product = %s',
-			_DB_PREFIX_,
-			$this->item->id
-		);
+		/**
+		 * check table
+		 */
+		if ($this->checkTable(sprintf('%sspecific_price', _DB_PREFIX_)))
+		{
+			$now    = date('Y-m-d H:i:s');
+			$select = sprintf('SELECT * from %sspecific_price WHERE id_product = %s AND %s AND %s AND (id_shop = %s OR id_shop = 0)', _DB_PREFIX_, $this->item->id, '(`from` = "0000-00-00 00:00:00" OR "'.$now.'" >= `from`)', '(`to` = "0000-00-00 00:00:00" OR "'.$now.'" <= `to`)', $this->context->shop->id);
 
-		return Db::getInstance()->ExecuteS($select);
+			return Db::getInstance()->ExecuteS($select);
+		}
+		else
+		{
+			/**
+			 * empty
+			 */
+			return array ();
+		}
+	}
+
+	/**
+	 * check table name
+	 *
+	 * @param $tableName
+	 *
+	 * @return bool
+	 */
+	protected function checkTable($tableName)
+	{
+		foreach (Db::getInstance()->ExecuteS('SHOW TABLES') as $key => $name)
+		{
+			if ($tableName == current($name))
+			{
+				/**
+				 * is current
+				 */
+				return true;
+			}
+		}
+
+		return false;
 	}
 
 	protected function getCategoriesFromDb()
 	{
-		$select = sprintf(
-			'SELECT
+		if (version_compare(_PS_VERSION_, '1.5.0.0', '>='))
+		{
+			$select = sprintf('SELECT
+						cp.id_category,
+						cp.position,
+						cl.name
+						FROM %scategory_product AS cp
+						LEFT JOIN %scategory_lang AS cl
+						ON cp.id_category = cl.id_category
+						LEFT JOIN %scategory_shop AS cs
+						ON cs.id_category = cp.id_category
+						WHERE cp.id_product = %s AND
+						cl.id_lang = %s AND
+						cs.id_shop = %s
+						group by cp.id_category
+						', _DB_PREFIX_, _DB_PREFIX_, _DB_PREFIX_, $this->item->id, $this->getContext()->language->id, $this->getContext()->shop->id);
```

Comment:
If your SQL request uses interger as parameters, why don't you use %d with ```(int)``` casts ?